toadr3/report_descriptor.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any
+
+from .exceptions import SchemaException
+from .values_map import parse_values_map
+
+
+class ReportDescriptor:
+    """An object that may be used to request a report from a VEN.
+
+    See OpenADR REST User Guide for detailed description of how configure a report request.
+    """
+
+    def __init__(self, data: dict):
+        if "payloadType" not in data:
+            raise SchemaException("Missing 'payloadType' in report descriptor schema.")
+
+        self._payload_type: str = data["payloadType"]
+        self._reading_type: str | None = None
+        self._units: str | None = None
+        self._targets: dict[str, Any] | None = None
+        self._aggregate: bool = False
+        self._start_interval: int = -1
+        self._num_intervals: int = -1
+        self._historical: bool = True
+        self._frequency: int = -1
+        self._repeat: int = 1
+
+        if "readingType" in data:
+            self._reading_type = data["readingType"]
```

Comment:
Why not 
```python
self._reading_type = data.get("readingType", None)
```
on line 18, and simmilarly for the other attributes?

Reply:
No good reason, will fix.

Reply:
I see. I followed a pattern from some of the other objects where the check is required. Will still fix these though.